docs/getting-started/publish-infrastructure.md

```diff
@@ -0,0 +1,560 @@
+  writeConnectionSecretsToNamespace: crossplane-system
+  reclaimPolicy: Delete
+  from:
+    apiVersion: database.example.org/v1alpha1
+    kind: MySQLInstance
+  to:
+    - base:
+        apiVersion: database.alibaba.crossplane.io/v1alpha1
+        kind: RDSInstance
+        spec:
+          forProvider:
+            engine: mysql
+            engineVersion: "10.0"
+            dbInstanceClass: rds.mysql.s1.small
+            dbInstanceStorageInGB: 20
+            securityIPList: "0.0.0.0/0"
+            masterUsername: "myuser"
+          writeConnectionSecretsToRef:
+            namespace: crossplane-system
+          providerRef:
+            name: alibaba-provider
+          reclaimPolicy: Delete
+      patches:
+        - fromFieldPath: "metadata.uid"
+          toFieldPath: "spec.writeConnectionSecretToRef.name"
+          transforms:
+            - type: string
+              string:
+              fmt: "%s-mysql"
+        - fromFieldPath: "spec.storageGB"
+          toFieldPath: "spec.forProvider.dbInstanceStorageInGB"
+      connectionDetails:
+        - fromConnectionSecretKey: username
+        - fromConnectionSecretKey: password
+        - fromConnectionSecretKey: endpoint
+```
+
+Create the `Composition`:
+```
+kubectl apply -f alibaba-mysqlinstance.yaml
+```
+
+## Create Requirement
+
+Now that we have defined our new type of infrastructure (`MySQLInstance`) and
+created at least one composition that can satisfy it, we can create a
+`MySQLInstanceRequirement` in the namespace of our choosing. In each
+`Composition` we defined we added a `provider: <name-of-provider` label. In the
+`MySQLInstanceRequirement` below we use the `compositionSelector` with
+`provider: aws` to match our AWS `Composition`. This can be adjusted to select
+the provider of your choosing.
+
+Create a file named `mysql-req.yaml` with the following contents:
+```yaml
+apiVersion: database.example.org/v1alpha1
+kind: MySQLInstanceRequirement
+metadata:
+  name: my-db
+  namespace: default
+spec:
+  parameters:
+    storageGB: 20
+  compositionSelector:
+    matchLabels:
+      provider: aws
+  writeConnectionSecretToRef:
+    name: db-conn
+```
+
+Create the `MySQLRequirement`:
+```
+kubectl apply -f mysql-req.yaml
+```
+
+Once the database is provisioned on your provider of choice, you should see
+`STATUS: Bound` in the output when you run:
+```
+kubectl get -f mysql-req.yaml
+```
+
+You should also see a `Secret` in the `default` namespace named `db-conn` that
+contains fields for `username`, `password`, and `endpoint`:
+```
+kubectl get secrets db-conn
+```
+
+## Consume Infrastructure
+
+Because connection secrets are written as a Kubernetes `Secret` they can easily
+be consumed by Kubernetes primitives. The most basic building block in
+Kubernetes is the `Pod`. Let's define a `Pod` that will show that we are able to
+connect to our newly provisioned database.
+
+Create a file named `pod.yaml` with the following contents:
+```yaml
+apiVersion: v1
+kind: Pod
```

Comment:
Hmm yeah I think that would make sense, except for I’m not sure if this `Pod` will get into a long `CrashLoopBackoff` waiting for the DB.

Reply:
If we create it after the DB is ready like composition example, why would it get into `CrashLoopBackoff`?

Reply:
Aren’t you saying applying it all as one set of YAML?

Reply:
I was saying that since Azure requires at least 3 resources for public access, we can deploy those in one big YAML. Others would stay same. After all is done, we can instruct user to create this `Pod` after they see all resources they deployed are ready.

Reply:
> I was saying that since Azure requires at least 3 resources for public access, we can deploy those in one big YAML. 

@muvaf I think I may not be following here. The Azure `Composition` has all 3 of these resources.

Reply:
I meant to add the `Pod` to the other document that is about provisioning infrastructure. The thing that would hold us back from that is Azure's extra resource requirement and we can create those together with MySQLServer, manually, as opposed to Composition.

Reply:
> Should we create this Pod in infra provisioning step, too?

No, I don't think we should. Let's keep that section super simple; you apply YAML, you get infra in your cloud console. In reality I'm not sure we should be encouraging folks to statically provision something like a MySQLInstance and use it from a pod without ever creating a `Requirement`.